app/components/primer/auto_complete.rb

```diff
@@ -1,88 +1,116 @@
 # frozen_string_literal: true
 
 module Primer
-  # Use `AutoComplete` to populate input values from server search results.
+  # Use `AutoComplete` to provide a user with a list of selectable suggestions that appear when they type into the
+  # input field. This list is populated by server search results.
   # @accessibility
-  #   Always provide a label for the `Autocomplete` component.
+  #   Always provide an accessible label to help the user interact with the input element and listbox popup.
+  #
+  #   To show a visible label, set the `label` slot.
+  #
+  #   If you do not wish to provide a visible label, you must set an `aria-label` attribute. You may set
+  #   `:"aria-label"` directly on `AutoComplete` instead of the slots and Primer will apply it to the correct elements.
   class AutoComplete < Primer::Component
     status :beta
 
     DEFAULT_INPUT_TYPE = :text
     INPUT_TYPE_OPTIONS = [DEFAULT_INPUT_TYPE, :search].freeze
 
+    # Optionally render a visible label. See <%= link_to_accessibility %>
+    #
+    # @param for [Symbol] id of input
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :label, lambda { |**system_arguments|
+      raise ArgumentError, "missing keyword: for" if !system_arguments.key?(:for) && !Rails.env.production?
+
+      @input_id_for_label = system_arguments[:for]
+      system_arguments[:tag] = :label
+      Primer::BaseComponent.new(**system_arguments)
+    }
+
     # Required input used to search for results
     #
     # @param type [Symbol] <%= one_of(Primer::AutoComplete::INPUT_TYPE_OPTIONS) %>
     # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
     renders_one :input, lambda { |type: DEFAULT_INPUT_TYPE, classes: "form-control", **system_arguments|
       system_arguments[:tag] = :input
+      system_arguments[:role] = :combobox
+
```

Comment:
Priority order:

- `<label>` associated with this input id
- `aria-label` set to this slot
- `aria-label` passed to parent